x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/optimizer/PhysicalPlanOptimizerTests.java

```diff
@@ -420,7 +419,7 @@ public void testExtractorMultiEvalWithSameName() {
         var extract = as(project.child(), FieldExtractExec.class);
         assertThat(
             names(extract.attributesToExtract()),
-            contains("_meta_field", "emp_no", "first_name", "gender", "job", "job.raw", "languages", "last_name", "salary")
```

Comment:
I don't think the test is reflecting the reality. The FieldExtractExec here should ask for `job` as well.

Reply:
In the end the list of outputted fields/columns includes both `job` and `job.raw` and they can be different. If `raw` is a sub-field of type `keyword` with `ignore_above: 5` it can have no value (nothing is indexed in that field), while its parent does.